### ISSUE TITLE
Size of parquet file larger than RUB

### ISSUE THREAD

**Issue by NGA-TRAN** [influxdata/influxdb_iox#2190]
While playing with chunks system table, I found a few parquet chunks smaller than their  in memory `ReadBufferAndObjectStore`. 

# Reproducer

1. Connect to Query-3000
  ```
 kubectl -n iox port-forward svc/iox-query-3000 1234:8082  1235:8080
  ./target/debug/influxdb_iox --host http://localhost:1234 sql
```

2. In sql interface, run observer to download system table Information first
`   observer;
`
3. Then run these queries

```
select count(*) from chunks where storage='ReadBufferAndObjectStore';
select count(*) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes < object_store_bytes;
select count(*) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 2;
select count(*) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 3;
select count(*) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 4;
select count(*) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 10;

select avg(row_count) from chunks where storage='ReadBufferAndObjectStore';
select avg(row_count) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes < object_store_bytes;
select avg(row_count) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 2;
select avg(row_count) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 3;
select avg(row_count) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 4;
select avg(row_count) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 10;

select memory_bytes, object_store_bytes, row_count from chunks where storage='ReadBufferAndObjectStore' and memory_bytes < object_store_bytes;
```

What I see from my runs:

```
OBSERVER> select count(*) from chunks where storage='ReadBufferAndObjectStore';
+-----------------+
| COUNT(UInt8(1)) |
+-----------------+
| 353             |
+-----------------+

Returned 1 row in 13.011828ms
OBSERVER> select count(*) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 2;
+-----------------+
| COUNT(UInt8(1)) |
+-----------------+
| 169             |
+-----------------+

Returned 1 row in 14.123982ms
OBSERVER> select count(*) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 3;
+-----------------+
| COUNT(UInt8(1)) |
+-----------------+
| 98              |
+-----------------+

Returned 1 row in 5.714007ms
OBSERVER> select count(*) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 4;
+-----------------+
| COUNT(UInt8(1)) |
+-----------------+
| 86              |
+-----------------+

Returned 1 row in 5.823861ms
OBSERVER> select count(*) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 5;
+-----------------+
| COUNT(UInt8(1)) |
+-----------------+
| 85              |
+-----------------+

Returned 1 row in 5.451819ms
OBSERVER> select count(*) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 10;
+-----------------+
| COUNT(UInt8(1)) |
+-----------------+
| 35              |
+-----------------+

Returned 1 row in 5.495419ms

OBSERVER>    select avg(row_count) from chunks where storage='ReadBufferAndObjectStore';
+-------------------+
| AVG(row_count)    |
+-------------------+
| 2377327.495750708 |
+-------------------+

Returned 1 row in 8.992956ms
OBSERVER> select count(*) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes < object_store_bytes;
+-----------------+
| COUNT(UInt8(1)) |
+-----------------+
| 12              |
+-----------------+

Returned 1 row in 6.213869ms
OBSERVER> select memory_bytes, object_store_bytes, row_count from chunks where storage='ReadBufferAndObjectStore' and memory_bytes < object_store_bytes;
+--------------+--------------------+-----------+
| memory_bytes | object_store_bytes | row_count |
+--------------+--------------------+-----------+
| 6190934      | 8646441            | 64205     |
| 6272916      | 8768997            | 64830     |
| 2382790      | 3312616            | 23839     |
| 6268813      | 8617085            | 63899     |
| 6262755      | 8615930            | 64287     |
| 6344885      | 8765655            | 65082     |
| 2406647      | 3353291            | 23860     |
| 6171200      | 8597490            | 63970     |
| 3399290      | 4679123            | 34079     |
| 6177985      | 8613352            | 63701     |
| 6272857      | 8779051            | 65021     |
| 2388680      | 3341388            | 23846     |
+--------------+--------------------+-----------+

Returned 12 rows in 4.329728ms

OBSERVER> 
select avg(row_count) from chunks where storage='ReadBufferAndObjectStore';
+-------------------+
| AVG(row_count)    |
+-------------------+
| 2377327.495750708 |
+-------------------+

Returned 1 row in 5.050987ms
OBSERVER> 
select avg(row_count) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes < object_store_bytes;
+----------------+
| AVG(row_count) |
+----------------+
| 51718.25       |
+----------------+

Returned 1 row in 5.863189ms
OBSERVER> 
select avg(row_count) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 2;
+-------------------+
| AVG(row_count)    |
+-------------------+
| 4893049.455621302 |
+-------------------+

Returned 1 row in 6.068432ms
OBSERVER> 
select avg(row_count) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 3;
+-------------------+
| AVG(row_count)    |
+-------------------+
| 8422077.285714285 |
+-------------------+

Returned 1 row in 6.053588ms
OBSERVER> 
select avg(row_count) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 4;
+-------------------+
| AVG(row_count)    |
+-------------------+
| 9597021.918604651 |
+-------------------+

Returned 1 row in 6.203586ms
OBSERVER> 
select avg(row_count) from chunks where storage='ReadBufferAndObjectStore' and memory_bytes > object_store_bytes * 10;
+-------------------+
| AVG(row_count)    |
+-------------------+
| 7439716.571428572 |
+-------------------+



```

**Comment by alamb** [comment]
## Summary
TLDR; @e-dard  read buffer compression is awesome and does indeed get better compression than these parquet files. However, IOx is currently writing parquet files without the final "block level compression" (aka snappy or zstd) as @pauldix  suspected, and once that is enabled the parquet file drops by half. 

Thus, my plan is to enable `zstd` block level compression in IOx when writing parquet files

## Worked Problem:


The basic plan was to identify a chunk that had a larger parquet file and go download the corresponding parquet file and see what was happening


```
844910ece80be8bc_3c0bd4c89186ca89> select id, partition_key, table_name, memory_bytes, object_store_bytes, row_count  from system.chunks where storage='ReadBufferAndObjectStore' and memory_bytes < object_store_bytes;
+----+---------------------+------------------------------------------------+--------------+--------------------+-----------+
| id | partition_key       | table_name                                     | memory_bytes | object_store_bytes | row_count |
+----+---------------------+------------------------------------------------+--------------+--------------------+-----------+
| 3  | 2021-08-06 15:00:00 | storage_producer_node_request_duration_seconds | 5723214      | 7712720            | 61562     |
| 3  | 2021-08-06 14:00:00 | storage_producer_node_request_duration_seconds | 5214451      | 7730506            | 62270     |
| 6  | 2021-08-06 14:00:00 | storage_producer_node_request_duration_seconds | 5890675      | 7876242            | 64217     |
| 8  | 2021-08-06 14:00:00 | storage_producer_node_request_duration_seconds | 2246415      | 2941313            | 23543     |
| 6  | 2021-08-06 13:00:00 | storage_producer_node_request_duration_seconds | 5306434      | 7395769            | 59448     |
| 6  | 2021-08-06 13:00:00 | task_executor_run_latency_seconds              | 547790       | 694491             | 15151     |
| 2  | 2021-08-06 14:00:00 | task_executor_run_latency_seconds              | 588872       | 768083             | 16303     |
| 8  | 2021-08-06 14:00:00 | task_executor_run_latency_seconds              | 299321       | 310570             | 5616      |
+----+---------------------+------------------------------------------------+--------------+--------------------+-----------+
```

Note the number of rows is relatively small (61k) and the RUB reports 5.7M and the parqeuet filereports 7.7M

Here is the table schema:

```
844910ece80be8bc_3c0bd4c89186ca89>     show columns in storage_producer_node_request_duration_seconds;
+---------------+--------------+------------------------------------------------+-----------------------+-----------------------------+-------------+
| table_catalog | table_schema | table_name                                     | column_name           | data_type                   | is_nullable |
+---------------+--------------+------------------------------------------------+-----------------------+-----------------------------+-------------+
| public        | iox          | storage_producer_node_request_duration_seconds | 0.001                 | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.0015                | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.0022500000000000003 | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.0033750000000000004 | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.005062500000000001  | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.0075937500000000015 | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.011390625000000001  | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.017085937500000002  | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.025628906250000003  | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.03844335937500001   | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.057665039062500006  | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.08649755859375001   | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.129746337890625     | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.1946195068359375    | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.29192926025390625   | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.43789389038085935   | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.656840835571289     | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 0.9852612533569336    | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 1.4778918800354004    | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 11.222741464018819    | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 16.83411219602823     | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 2.2168378200531005    | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 3.3252567300796505    | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 4.987885095119475     | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | 7.481827642679213     | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | count                 | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | env                   | Dictionary(Int32, Utf8)     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | host                  | Dictionary(Int32, Utf8)     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | hostname              | Dictionary(Int32, Utf8)     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | nodename              | Dictionary(Int32, Utf8)     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | role                  | Dictionary(Int32, Utf8)     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | status                | Dictionary(Int32, Utf8)     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | sum                   | Float64                     | YES         |
| public        | iox          | storage_producer_node_request_duration_seconds | time                  | Timestamp(Nanosecond, None) | NO          |
| public        | iox          | storage_producer_node_request_duration_seconds | url                   | Dictionary(Int32, Utf8)     | YES         |
+---------------+--------------+------------------------------------------------+-----------------------+-----------------------------+-------------+
```

Here are the statistics (note the very small number of bytes consumed by the smaller integer values == @e-dard 's  ReadBuffer compression is pretty sweet)

```
844910ece80be8bc_3c0bd4c89186ca89>  select column_name, row_count, null_count, min_value, max_value, memory_bytes from system.chunk_columns where table_name = 'storage_producer_node_request_duration_seconds' AND chunk_id=3 AND partition_key='2021-08-06 15:00:00';
+-----------------------+-----------+------------+-----------------------------------------+--------------------------------------------------+--------------+
| column_name           | row_count | null_count | min_value                               | max_value                                        | memory_bytes |
+-----------------------+-----------+------------+-----------------------------------------+--------------------------------------------------+--------------+
| 0.001                 | 61562     | 0          | 0                                       | 18                                               | 1027         |
| 0.0015                | 61562     | 0          | 0                                       | 40                                               | 2539         |
| 0.0022500000000000003 | 61562     | 0          | 0                                       | 152                                              | 7491         |
| 0.0033750000000000004 | 61562     | 0          | 0                                       | 496                                              | 22019        |
| 0.005062500000000001  | 61562     | 0          | 0                                       | 1222                                             | 47707        |
| 0.0075937500000000015 | 61562     | 0          | 0                                       | 2644                                             | 75763        |
| 0.011390625000000001  | 61562     | 0          | 0                                       | 5435                                             | 100003       |
| 0.017085937500000002  | 61562     | 0          | 0                                       | 9616                                             | 125363       |
| 0.025628906250000003  | 61562     | 0          | 0                                       | 15765                                            | 158339       |
| 0.03844335937500001   | 61562     | 0          | 0                                       | 25083                                            | 115821       |
| 0.057665039062500006  | 61562     | 0          | 0                                       | 48047                                            | 128485       |
| 0.08649755859375001   | 61562     | 0          | 0                                       | 113323                                           | 195253       |
| 0.129746337890625     | 61562     | 0          | 0                                       | 841433                                           | 246395       |
| 0.1946195068359375    | 61562     | 0          | 0                                       | 849227                                           | 246395       |
| 0.29192926025390625   | 61562     | 0          | 0                                       | 860142                                           | 246395       |
| 0.43789389038085935   | 61562     | 0          | 0                                       | 862822                                           | 246395       |
| 0.656840835571289     | 61562     | 0          | 0                                       | 863072                                           | 246395       |
| 0.9852612533569336    | 61562     | 0          | 0                                       | 863318                                           | 246395       |
| 1.4778918800354004    | 61562     | 0          | 0                                       | 863535                                           | 246395       |
| 11.222741464018819    | 61562     | 0          | 1                                       | 864253                                           | 246395       |
| 16.83411219602823     | 61562     | 0          | 1                                       | 864530                                           | 246395       |
| 2.2168378200531005    | 61562     | 0          | 1                                       | 863704                                           | 246395       |
| 3.3252567300796505    | 61562     | 0          | 1                                       | 863830                                           | 246395       |
| 4.987885095119475     | 61562     | 0          | 1                                       | 863920                                           | 246395       |
| 7.481827642679213     | 61562     | 0          | 1                                       | 864098                                           | 246395       |
| count                 | 61562     | 0          | 1                                       | 864898                                           | 246395       |
| env                   | 61562     | 0          | monitor301-us-west-1                    | stag101-us-east-1                                | 3672         |
| host                  | 61562     | 0          | gateway-external-write-54575f94dc-5frxz | gateway-internal-write-bdd9c8779-vfvs7           | 110730       |
| hostname              | 61562     | 0          | gateway-external-write-54575f94dc-5frxz | gateway-internal-write-bdd9c8779-vfvs7           | 110730       |
| nodename              | 61562     | 0          | aks-storage1-23576596-vmss000016        | ip-10-156-97-220.ap-southeast-2.compute.internal | 40374        |
| role                  | 61562     | 0          | gateway-external-write                  | gateway-internal-write                           | 1098         |
| status                | 61562     | 0          | error-kafka-waiter                      | ok                                               | 954          |
| sum                   | 61562     | 0          | 0.104233414                             | 105247.16904445134                               | 492643       |
| time                  | 61562     | 0          | 1628262000000000000                     | 1628263500000000000                              | 492643       |
| url                   | 61562     | 0          | http://127.0.0.1:9999/metrics           | http://127.0.0.1:9999/metrics                    | 948          |
+-----------------------+-----------+------------+-----------------------------------------+--------------------------------------------------+--------------+
```

I fetched the parquet file from S3 using the AWS console (finding it in the directory structure was nice and easy, BTW -- props to @crepererum  and others who made this happen).

```
3000/844910ece80be8bc_3c0bd4c89186ca89/data/storage_producer_node_request_duration_seconds/2021-08-06 15:00:00/3.595452a2-f71b-4222-8bae-a6bde262d07e.parquet
```

Here are the per-column storage breakdown in parquet

```shell
alamb@ip-10-0-0-124 influxdb_iox % parquet-tools column-size  ~/Downloads/3.595452a2-f71b-4222-8bae-a6bde262d07e.parquet
parquet-tools column-size  ~/Downloads/3.595452a2-f71b-4222-8bae-a6bde262d07e.parquet
role                   -> Size In Bytes: 395
0.03844335937500001    -> Size In Bytes: 135739
0.017085937500000002   -> Size In Bytes: 79690
0.001                  -> Size In Bytes: 687
0.656840835571289      -> Size In Bytes: 442506
sum                    -> Size In Bytes: 552473
0.011390625000000001   -> Size In Bytes: 53808
0.29192926025390625    -> Size In Bytes: 440380
3.3252567300796505     -> Size In Bytes: 442640
7.481827642679213      -> Size In Bytes: 443760
hostname               -> Size In Bytes: 19639
host                   -> Size In Bytes: 19639
0.005062500000000001   -> Size In Bytes: 22147
0.0033750000000000004  -> Size In Bytes: 14204
0.0015                 -> Size In Bytes: 1569
0.0022500000000000003  -> Size In Bytes: 5499
0.129746337890625      -> Size In Bytes: 423940
count                  -> Size In Bytes: 443344
0.9852612533569336     -> Size In Bytes: 442978
1.4778918800354004     -> Size In Bytes: 442696
0.057665039062500006   -> Size In Bytes: 168619
env                    -> Size In Bytes: 925
4.987885095119475      -> Size In Bytes: 444048
url                    -> Size In Bytes: 429
16.83411219602823      -> Size In Bytes: 443232
nodename               -> Size In Bytes: 10795
0.1946195068359375     -> Size In Bytes: 436741
0.025628906250000003   -> Size In Bytes: 107205
0.0075937500000000015  -> Size In Bytes: 35561
time                   -> Size In Bytes: 65864
2.2168378200531005     -> Size In Bytes: 442768
0.43789389038085935    -> Size In Bytes: 440563
11.222741464018819     -> Size In Bytes: 443952
0.08649755859375001    -> Size In Bytes: 218265
status                 -> Size In Bytes: 227
```

It is pretty cool that some of the columns are legimiately larger in parquet than Edd's fancy read buffer encoding

However, looking at the metadata in the parquet file, it turns out each column is "uncompressed" (aka not run through a final block level encoder). It is also interesting to note, intellectually,  that there are multiple row groups in this parquet file with 21509,  21456, and  18597 rows respectively.

```shell
alamb@ip-10-0-0-124 influxdb_iox % parquet-tools meta   ~/Downloads/3.595452a2-f71b-4222-8bae-a6bde262d07e.parquet
parquet-tools meta   ~/Downloads/3.595452a2-f71b-4222-8bae-a6bde262d07e.parquet
file:                  file:/Users/alamb/Downloads/3.595452a2-f71b-4222-8bae-a6bde262d07e.parquet
creator:               parquet-rs version 5.1.0
extra:                 ARROW:schema = /////zQU ..
extra:                 IOX:metadata = CAUSDgl4VQ1h ....

file schema:           arrow_schema
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
0.001:                 OPTIONAL DOUBLE R:0 D:1
0.0015:                OPTIONAL DOUBLE R:0 D:1
...
sum:                   OPTIONAL DOUBLE R:0 D:1
time:                  REQUIRED INT64 L:TIMESTAMP(NANOS,false) R:0 D:0
url:                   OPTIONAL BINARY L:STRING R:0 D:1

row group 1:           RC:21509 TS:2519901 OFFSET:4
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
0.001:                  DOUBLE UNCOMPRESSED DO:4 FPO:164 SZ:427/427/1.00 VC:21509 ENC:RLE_DICTIONARY,PLAIN,RLE ST:[min: -0.0, max: 18.0, num_nulls not defined]
0.0015:                 DOUBLE UNCOMPRESSED DO:491 FPO:803 SZ:820/820/1.00 VC:21509 ENC:RLE_DICTIONARY,PLAIN,RLE ST:[min: -0.0, max: 38.0, num_nulls not defined]
...
sum:                    DOUBLE UNCOMPRESSED DO:2336484 FPO:2468112 SZ:162828/162828/1.00 VC:21509 ENC:RLE_DICTIONARY,PLAIN,RLE ST:[min: 0.110464962, max: 57838.74271707724, num_nulls not defined]
time:                   INT64 UNCOMPRESSED DO:2499379 FPO:2500908 SZ:23130/23130/1.00 VC:21509 ENC:RLE_DICTIONARY,PLAIN,RLE ST:[min: 2021-08-06T15:00:00.000000000, max: 2021-08-06T15:25:00.000000000, num_nulls not defined]
url:                    BINARY UNCOMPRESSED DO:2522577 FPO:2522624 SZ:143/143/1.00 VC:21509 ENC:RLE_DICTIONARY,PLAIN,RLE ST:[min: http://127.0.0.1:9999/metrics, max: http://127.0.0.1:9999/metrics, num_nulls not defined]

row group 2:           RC:21456 TS:2788784 OFFSET:2522827
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
0.001:                  DOUBLE UNCOMPRESSED DO:2522827 FPO:2522889 SZ:184/184/1.00 VC:21456 ENC:RLE_DICTIONARY,PLAIN,RLE ST:[min: -0.0, max: 5.0, num_nulls not defined]
0.0015:                 DOUBLE UNCOMPRESSED DO:2523078 FPO:2523342 SZ:659/659/1.00 VC:21456 ENC:RLE_DICTIONARY,PLAIN,RLE ST:[min: -0.0, max: 40.0, num_nulls not defined]
...
sum:                    DOUBLE UNCOMPRESSED DO:5079448 FPO:5251020 SZ:211899/211899/1.00 VC:21456 ENC:RLE_DICTIONARY,PLAIN,RLE ST:[min: 263.0652116830002, max: 105247.16904445134, num_nulls not defined]
time:                   INT64 UNCOMPRESSED DO:5291414 FPO:5292719 SZ:22850/22850/1.00 VC:21456 ENC:RLE_DICTIONARY,PLAIN,RLE ST:[min: 2021-08-06T15:00:00.000000000, max: 2021-08-06T15:25:00.000000000, num_nulls not defined]
url:                    BINARY UNCOMPRESSED DO:5314332 FPO:5314379 SZ:143/143/1.00 VC:21456 ENC:RLE_DICTIONARY,PLAIN,RLE ST:[min: http://127.0.0.1:9999/metrics, max: http://127.0.0.1:9999/metrics, num_nulls not defined]

...
```


Then I used `parquet-tools` to apply various compressions:


```shell
parquet-tools trans-compression   ~/Downloads/3.595452a2-f71b-4222-8bae-a6bde262d07e.parquet  ~/Downloads/3.595452a2-f71b-4222-8bae-a6bde262d07e.snappy.parquet  SNAPPY
```

After which the parquet file is compressed down to 5.3MB (smaller than reported by the ReadBuffer)

```
parquet-tools size   ~/Downloads/3.595452a2-f71b-4222-8bae-a6bde262d07e.snappy.parquet
Total Size: 5373005 bytes
```

zstd got even better:

```
parquet-tools size  ~/Downloads/3.595452a2-f71b-4222-8bae-a6bde262d07e.zstd.parquet
Total Size: 3587704 bytes
```

So my plan will be to enable zstd compression for writing parquet files on IOx and call it a day.

**Comment by e-dard** [comment]
@alamb would it be possible before calling it a day to see the output of `parquet-tools column-size ` after applying the block compression, or does `parquet-tools column-size` always report the size once "block decompressed"? 

I guess put another way - is the block compression per column or per page/row group etc?

**Comment by NGA-TRAN** [comment]
Nice discovery, @alamb

**Comment by alamb** [comment]
> I guess put another way - is the block compression per column or per page/row group etc?

@e-dard  the answer s "per column" -- you can control the compression selected using this API in the rust implmentation

https://docs.rs/parquet/5.1.0/parquet/file/properties/struct.WriterPropertiesBuilder.html#method.set_column_compression

**Comment by alamb** [comment]
Here is the parquet file in case anyone else wants to mess around with it: 
[chunk_3.zip](https://github.com/influxdata/influxdb_iox/files/6947393/chunk_3.zip)
